rdkit/ML/Cluster/Butina.py

```diff
@@ -59,57 +65,83 @@ def ClusterData(data, nPts, distThresh, isDistData=False, distFunc=EuclideanDist
          The first element for each cluster is its centroid.
 
   """
-  if isDistData and len(data) > (nPts * (nPts - 1) / 2):
-    logger.warning("Distance matrix is too long")
-  nbrLists = [None] * nPts
-  for i in range(nPts):
-    nbrLists[i] = []
-
-  dmIdx = 0
-  for i in range(nPts):
-    for j in range(i):
-      if not isDistData:
-        dij = distFunc(data[i], data[j])
-      else:
-        dij = data[dmIdx]
-        dmIdx += 1
-      if dij <= distThresh:
-        nbrLists[i].append(j)
-        nbrLists[j].append(i)
-  # sort by the number of neighbors:
-  tLists = [(len(y), x) for x, y in enumerate(nbrLists)]
-  tLists.sort(reverse=True)
-
-  res = []
-  seen = [0] * nPts
-  while tLists:
-    _, idx = tLists.pop(0)
+  if isDistData:
+    # Check if data is a supported type
+    if not isinstance(data, (list, np.ndarray)):
+      raise ValueError(f"Unsupported type for data, {type(data)}")
+    
+    # Check if data is a 1D array or list
+    if isinstance(data, list) or isinstance(data, np.ndarray) and data.ndim == 1:
```

Comment:
readability suggestion:
```suggestion
    if isinstance(data, list) or (isinstance(data, np.ndarray) and data.ndim == 1):
```